Data/ApplicationDbContext.cs

```diff
@@ -63,6 +72,119 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
+                .IsRequired();
+
+            builder.HasMany(m => m.Images)
+                .WithOne(i => i.Move)
+                .HasForeignKey(i => i.MoveId);
+
+            builder.HasMany(m => m.Trainings)
+                .WithMany(t => t.Moves)
+                .UsingEntity<Dictionary<string, object>>(
+                    "MoveTraining",
+                    j => j.HasOne<Training>().WithMany().HasForeignKey("TrainingId"),
+                    j => j.HasOne<Move>().WithMany().HasForeignKey("MoveId"));
+        }
+    }
+
+    public class ImageConfiguration : IEntityTypeConfiguration<Image>
+    {
+        public void Configure(EntityTypeBuilder<Image> builder)
+        {
+            builder.HasKey(i => i.Id);
+
+            builder.Property(i => i.Name)
+                .IsRequired()
+                .HasColumnType("nvarchar(50)");
+
+            builder.Property(i => i.ImageData)
+                .IsRequired()
+                .HasColumnType("varbinary(max)");
+        }
+    }
+
+    public class TrainingConfiguration : IEntityTypeConfiguration<Training>
+    {
+        public void Configure(EntityTypeBuilder<Training> builder)
+        {
+            builder.HasKey(t => t.Id);
+
+            builder.Property(t => t.Name)
+                .IsRequired()
+                .HasColumnType("nvarchar(50)");
+
+            builder.Property(t => t.Description)
+                .IsRequired()
+                .HasColumnType("nvarchar(1000)");
+
+            builder.Property(t => t.Type)
+                .IsRequired()
+                .HasColumnType("nvarchar(50)");
+
+            builder.Property(t => t.ImageUrl)
+                .IsRequired(false);
+
+            builder.HasMany(t => t.Moves)
+                .WithMany(m => m.Trainings)
+                .UsingEntity<Dictionary<string, object>>(
+                    "MoveTraining",
+                    j => j.HasOne<Move>().WithMany().HasForeignKey("MoveId"),
+                    j => j.HasOne<Training>().WithMany().HasForeignKey("TrainingId"));
+        }
+    }
+
+    public class ClassConfiguration : IEntityTypeConfiguration<Class>
+    {
+        public void Configure(EntityTypeBuilder<Class> builder)
+        {
+            builder.HasKey(c => c.Id);
+
+            builder.Property(c => c.Duration)
+                .IsRequired()
+                .HasColumnType("nvarchar(50)");
+
+            builder.Property(c => c.Date)
+                .IsRequired();
+
+            builder.Property(c => c.Location)
+                .IsRequired()
+                .HasColumnType("nvarchar(50)");
+
+            builder.Property(c => c.Trainer)
+                .IsRequired()
+                .HasColumnType("nvarchar(50)");
+
+            builder.HasOne(c => c.Training)
+                .WithMany(t => t.Classes)
+                .HasForeignKey(c => c.TrainingId)
+                .OnDelete(DeleteBehavior.Cascade);
+
+            builder.Property(c => c.RegisteredUsers)
```

Comment:
I jak to zapisujesz w bazie danych jako string - jako varchar i tam lista username-ów username1,username2..etc ? Hmm to może i zadziała, ale jest średnie implementacyjnie